### DISCUSSION TITLE
Automatically add a person to a membership

### DISCUSSION THREAD

**Discussion by carlbannister** [liedekef/events-made-easy#396] [Q&A] [answered]
Hi

We have a basic tier of membership that everybody needs to be a member of to participate in an event. Is there a way that when a person fills in an event form, they automatically become a member of the basic membership without them having to apply for membership first?

Thanks

**Comment by liedekef** [comment]
EME allows you to add peope to a group, but not to a membership. The way you describe it, you require a membership to be able to sign up for an event, but you want people to automatically become a member when they sign up ... that's weird ...

**Comment by carlbannister** [comment]
Yes kind of weird, we used to have members with the usual membership
process and discounted event fee for being so. Then we had non members who
paid a full event fee on the day.
We are now forced that everyone that attends is a member which is why for
the casual attendees I would like to add them to a  membership giving basic
member rights with no discount when they sign up to an event.
Backwards I know but logical for our setup

**Reply by liedekef** [reply]
If EME checks for a membership, then if not a member access is denied (not allowed and become a member of). So what you want is simply not possible. I would go for the group-solution: for those attending an event (that doesn't require the group) you can automatically add them to the group (and even make a WP user).
Otherwise you'll need to investigate the EME filters/hooks ... and calling the function eme_add_update_member (which checks nonces, if the person exists, if the person is already a member etc ...)

**Reply by carlbannister** [reply, marked as answer]
Great thanks for the reply, maybe an easier way would be just to set all events to look for membership and then on the event booking page just put a link to the basic membership page. Is it possible to put a link on an event form to a membership page, (sorry if  a dumb question to you but i have very limited programming knowledge and learning as i go along)

**Reply by carlbannister** [reply]
Scrap that question, have figured out .

Thanks for the great and continued support
